mirdata/datasets/saraga.py

```diff
@@ -0,0 +1,621 @@
+        )
+        self.metadata_path = utils.none_path_join(
+            [self._data_home, self._track_paths['metadata'][0]]
+        )
+
+        # Flag to separate between carnatinc and hindustani tracks
+        self.iam_style = str(self.track_id.split('_')[0])
+
+        # CARNATIC MUSIC TRACKS
+        if self.iam_style == 'carnatic':
+            metadata = DATA.metadata(self.metadata_path)
+            if metadata is not None and track_id == metadata['track_id']:
+                self._track_metadata = metadata
+            else:
+                # annotations with missing metadata
+                self._track_metadata = {
+                    'raaga': None,
+                    'form': None,
+                    'title': None,
+                    'work': None,
+                    'length': None,
+                    'taala': None,
+                    'album_artists': None,
+                    'mbid': None,
+                    'artists': None,
+                    'concert': None
+                }
+
+            self.raaga = self._track_metadata['raaga']
+            self.form = self._track_metadata['form']
+            self.title = self._track_metadata['title']
+            self.work = self._track_metadata['work']
+            self.taala = self._track_metadata['taala']
+            self.album_artists = self._track_metadata['album_artists']
+            self.mbid = self._track_metadata['mbid']
+            self.artists = self._track_metadata['artists']
+            self.concert = self._track_metadata['concert']
+
+        # HINDUSTANI MUSIC TRACKS
+        if self.iam_style == 'hindustani':
+            metadata = DATA.metadata(self.metadata_path)
+            if metadata is not None and track_id == metadata['track_id']:
+                self._track_metadata = metadata
+            else:
+                # annotations with missing metadata
+                self._track_metadata = {
+                    'title': None,
+                    'raags': None,
+                    'length': None,
+                    'album_artists': None,
+                    'forms': None,
+                    'mbid': None,
+                    'artists': None,
+                    'release': None,
+                    'works': None,
+                    'taals': None,
+                    'layas': None
+                }
+
+            self.title = self._track_metadata['title']
+            self.raags = self._track_metadata['raags']
+            self.album_artists = self._track_metadata['album_artists']
+            self.forms = self._track_metadata['forms']
+            self.mbid = self._track_metadata['mbid']
+            self.artists = self._track_metadata['artists']
+            self.release = self._track_metadata['release']
+            self.works = self._track_metadata['works']
+            self.taals = self._track_metadata['taals']
+            self.layas = self._track_metadata['layas']
```

Comment:
Not sure about having different available metadata depending on a flag - could you instead have all the same metadata but fill it differently depending on a flag, using `None` when the data isn't available for that track?

for e.g. `self.layas` = None for Carnatic tracks and `self.layas = self._track_metadata['layas']` for hindustani tracks.

Reply:
The issue here is that individual metadata files for tracks have also different dicts depending on the IAM style, Carnatic or Hindustani. So the specific metadata attributes of Carnatic tracks do not exist in Hindustani metadata files. 

Metadata files are coded in json files, so the metadata is loaded with a single line: `metadata = json.load(f)`, where `f` is the json parser. So when metadata exists, all the fields or attributes from the other IAM style will not exist, instead of being None. What I can do is rewritting the `_load_metadata` function so it creates a metadata dictionary with all the attributes from both Carnatic and Hindustani tracks and it just fills de fields of the IAM Style of each certain track. What are your thoughts about this?

Reply:
Yes I see what you mean! My suggestion is to merge lines 183-191 + 214-223, outside of the if statements, and setting the possibly missing attributes with `None` when they don't apply.

Reply:
Ok, I understand! Fixing it now:)